vkcs/mlplatform/resource_spark_k8s.go

```diff
@@ -0,0 +1,502 @@
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"suspend_after_inactive_min": schema.Int64Attribute{
+				Optional:    true,
+				Description: "Timeout of cluster inactivity before suspending. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+			},
+
+			"delete_after_inactive_min": schema.Int64Attribute{
+				Optional:    true,
+				Description: "Timeout of cluster inactivity before deletion. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.RequiresReplace(),
+				},
+			},
+
+			"s3_bucket_name": schema.StringAttribute{
+				Computed:    true,
+				Description: "S3 bucket name",
+			},
+
+			"history_server_url": schema.StringAttribute{
+				Computed:    true,
+				Description: "URL of the history server",
+			},
+
+			"control_instance_id": schema.StringAttribute{
+				Computed:    true,
+				Description: "ID of the control instance",
+			},
+
+			"inactive_min": schema.Int64Attribute{
+				Computed:    true,
+				Description: "Minutes of cluster inactivity",
+			},
+
+			"region": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "The `region` to fetch availability zones from, defaults to the provider's `region`.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplaceIf(planmodifiers.GetRegionPlanModifier(resp),
+						"require replacement if configuration value changes", "require replacement if configuration value changes"),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+
+			"timeouts": timeouts.Attributes(ctx, timeouts.Opts{
+				Create: true,
+				Update: true,
+				Delete: true,
+			}),
+		},
+		Description: "Manages a ML Platform Spark K8S cluster resource.",
+	}
+}
+
+func (r *SparkK8SResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.config = req.ProviderData.(clients.Config)
+}
+
+func (r *SparkK8SResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data SparkK8SResourceModel
+	diags := req.Plan.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	timeout, diags := data.Timeouts.Create(ctx, clusterCreateTimeout)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	region := data.Region.ValueString()
+	if region == "" {
+		region = r.config.GetRegion()
+	}
+
+	mlPlatformClient, err := r.config.MLPlatformV1Client(region)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating VKCS ML Platform client", err.Error())
+		return
+	}
+
+	nodeGroups := expandNodeGroupOpts(data.NodeGroups)
+
+	sparkConfiguration := data.SparkConfiguration.ValueString()
+	sparkConfigurationEncoded := base64.StdEncoding.EncodeToString([]byte(sparkConfiguration))
```

Comment:
Maybe it's better to place this logic in the opts builder. It will be easier to debug when we can see underlying values instead of base64 string in the most of places. This also reduces code's repeatability.

Reply:
Moved logic